clients/drcachesim/drpt2trace/pt2ir.cpp

```diff
@@ -287,6 +290,15 @@ pt2ir_t::convert(DR_PARAM_IN const uint8_t *pt_data, DR_PARAM_IN size_t pt_data_
     /* This flag indicates whether manual synchronization is required. */
     bool manual_sync = true;
 
+    uint64_t decoded_instr_count = 0;
+    uint64_t non_fatal_decode_error_count = 0;
+    /* XXX: This is currently set based on empirical observations. We use this heuristic
+     * to detect errors where we can still produce a trace for this syscall albeit with
+     * some PC discontinuities. Specifically: we allow MAX_ERROR_COUNT non-consecutive
+     * errors of type pte_bad_query.
```

Comment:
Where do you end up missing an entire syscall?  Is it code not visible in this diff view?  But doesn't it need to be connected to the new flag?

Reply:
That's in raw2trace. If pt2ir.convert returns a non-success error code that signals failure in PT data decode, we let the syscall drop. It is connected to the new flag.